lib/temperature_monitor/temperature_handler.ex

```diff
@@ -0,0 +1,91 @@
+defmodule TemperatureMonitor.TemperatureHandler do
+  use GenServer
+
+  def start_link(stash_pid) do
+    {:ok, _pid} = GenServer.start_link(__MODULE__, stash_pid, name: __MODULE__)
+  end
+
+  def init(stash_pid) do
+    filelist = TemperatureMonitor.Stash.get_value stash_pid
+    monitored_paths = []
+     {:ok, {filelist, stash_pid, monitored_paths}}
+  end
+
+  def callback(:stop) do
+    IO.puts "STOP"
+  end
+
+  def callback(filepath, events) do
+    IO.inspect {filepath, events}
+  end
+
+  def monitor_path(base_path) do
+    GenServer.cast(__MODULE__, {:add_base_path, base_path})
+  end
+
+  def get_monitored_paths do
```

Comment:
Does the `base_path` ever change during runtime? If it is a fixed value for life of the process then instead of managing it in a `GenServer` it might be better to just have that be a concern for the `Monitor` module and let this actor only worry about managing the list of temperature files.